File: communicate-server/src/main/java/com/knilim/knilim/server/MessageServer.java
package com.knilim.knilim.server;

import android.util.Log;

import java.util.Locale;

public class MessageServer {

    private static String TAG = "socket-io";

    /**
     * 单例实现
     */
    private static volatile MessageServer INSTANCE = null;
    public MessageServer getINSTANCE(String host, Integer port, MsgRcvHandler msgRcvHandler) {
        if(INSTANCE == null) {
            synchronized (MessageServer.class) {
                if(INSTANCE == null) {
                    INSTANCE = new MessageServer(host, port, msgRcvHandler);
                }
            }
        }
        return INSTANCE;
    }


    private String host;
    private Integer port;
    private MsgRcvHandler msgRcvHandler;

    private MessageServer(String host, Integer port, MsgRcvHandler msgRcvHandler) {
        this.host = host;
        this.port = port;
        this.msgRcvHandler = msgRcvHandler;
    }

    public boolean connect() {
        String url = String.format(Locale.US,"http://%s:%d/sockets", host, port);
        try {
           return true;
        } catch (Exception e) {
            Log.e(TAG, e.getMessage());
            return false;
        }
    }

    public boolean sendMessage() {
        try {
            return true;
        } catch (Exception e) {
            Log.e(TAG, e.getMessage());
            return false;
        }
    }
}

File: communicate-server/src/main/java/com/knilim/knilim/server/MessageServer.kt
package com.knilim.knilim.server

import android.util.Log
import java.util.*

object MessageServer {

    private const val TAG = "MessageServer"

    lateinit var host : String
    var port : Int? = null
    lateinit var msgRcvHandler: MsgRcvHandler

    fun config(host : String, port : Int, msgRcvHandler: MsgRcvHandler) {
        this.host = host
        this.port = port
        this.msgRcvHandler = msgRcvHandler
    }

    fun connect(): Boolean {
        val url = "http://${host}:${port}/sockets"
        return try {
            true
        } catch (e: Exception) {
            Log.e(TAG, e.message!!)
            false
        }
    }

    fun sendMessage(): Boolean {
        return try {
            true
        } catch (e: Exception) {
            Log.e(TAG, e.message!!)
            false
        }
    }
}